### PR TITLE
Add support for adding and clearing multi-valued configuration

### LINKED ISSUE #1719: Add API for unsetting multivar entry
## Reproduction steps

Given a multivar setting (i.e. "plugin"):

```
// throws LibGit2Sharp.LibGit2SharpException: 'entry is not unique due to being a multivar'
repo.Config.Unset("plugin");
```

### Expected behavior

Provide an `UnsetAll(key, level)` pair of overloads that basically expose the existing [UnsetMultivar](https://github.com/libgit2/libgit2sharp/blob/master/LibGit2Sharp/Configuration.cs#L256) behavior.

### Actual behavior

Exception thrown: `LibGit2Sharp.LibGit2SharpException: 'entry is not unique due to being a multivar'`

### Version of LibGit2Sharp (release number or SHA1)

0.26.1

### Operating system(s) tested; .NET runtime tested

.NET Core 2.1

### PR DESCRIPTION
**Pull request by kzu**
Implement value adding by exposing the underlying `set_multivar`,
which supports only string values in the underlying libgit2, so no
other typed overloads are provided at this point.

The counterpart for deleting keys exposes the underlying `delete_multivar`.
No regex-based overload is exposed for consistency with the existing `Set<T>`
overloads which don't expose it either.

Also exposed the boolean return value from the `Unset` calls which
is already present in the Proxy API.

Fixes #1719.

### CODE REVIEW

**Comment by kzu** [comment]
All green! 👍 

image

**Comment by kzu** [comment]
heya @bording this PR also gets travis back to green ;)

File: LibGit2Sharp/Configuration.cs [hunk 1]
```diff
@@ -233,33 +233,47 @@ public void Dispose()
...
             using (ConfigurationHandle h = RetrieveConfigurationHandle(level, true, configHandle))
             {
-                Proxy.git_config_delete(h, key);
+                return Proxy.git_config_delete(h, key);
             }
         }
 
-        internal void UnsetMultivar(string key, ConfigurationLevel level)
+        /// <summary>
+        /// Unset a configuration values in a multivar variable (key and value) in the local configuration.
```
Comment (bording) [inline LibGit2Sharp/Configuration.cs]:
```suggestion
        /// Unset all configuration values in a multivar variable (key and value) in the local configuration.
```

**Comment by bording** [comment]
This change is available in 0.27.0-preview-0024 on nuget.org.

**Pull request merged**